### PR TITLE
migrate to mongo DB (GDEV-1321)

### PR DESCRIPTION
**Pull request by KerstenBreuer**
```
- restructure core into hexagonal architecture
- replaced PostgreSQL-based DAO by MongoDB-based
   protocol/provider from hexkit
- migrated object storage abstraction from the chassis lib
   to hexkit
- switched to async python
- included api level conformance tests
```

### CODE REVIEW

File: dcs/ports/inbound/data_repository.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    async def access_drs_object(self, *, drs_id: str) -> models.DrsObjectWithAccess:
+        """
+        Serve the specified DRS object with access information.
+        If it does not exists in the outbox, yet, a RetryAccessLaterError is raised that
+        instructs to retry the call after a specified amount of time.
+        """
+        ...
+
+    @abstractmethod
+    async def registered_new_file(self, *, file: models.FileToRegister):
```
Comment (mephenor) [inline dcs/ports/inbound/data_repository.py:58]:
```suggestion
    async def register_new_file(self, *, file: models.FileToRegister):
```

**Review by mephenor — APPROVED** [review]
Looks good, just a few comments

File: setup.cfg [hunk 2]
```diff
@@ -35,6 +35,7 @@ zip_safe = False
 include_package_data = True
 packages = find:
 install_requires =
+    hexkit[mongodb,s3]==0.5.1
     ghga-service-chassis-lib[api,postgresql,s3]==0.15.0
```
Comment (MoritzHahn1337) [inline setup.cfg:39]: Do we still need the chassis lib?

Comment (mephenor) [reply setup.cfg:39]: Probably, but only the API part, right?

**Review by MoritzHahn1337 — APPROVED** [review]

**Review by mephenor — APPROVED** [review]

**Pull request merged**